client/src/pages/spools/create.tsx

```diff
@@ -105,17 +113,83 @@ export const SpoolCreate: React.FC<IResourceComponentsProps & CreateOrCloneProps
             }
           />
         </Form.Item>
+        <Form.Item
+          hidden={true}
+          name={["used_weight"]}
+          initialValue={0}
+        >
+          <InputNumber
+            value={used_weight}
+          />
+        </Form.Item>
+
         <Form.Item
           label={t("spool.fields.used_weight")}
           help={t("spool.fields_help.used_weight")}
-          name={["used_weight"]}
-          rules={[
-            {
-              required: true,
-            },
-          ]}
+          // name={["used_weight"]}
+          initialValue={0}
+        >
+          <InputNumber
+            min={0}
+            addonAfter="g"
+            precision={1}
+            formatter={numberFormatter}
+            parser={numberParser}
+            value={used_weight}
+            onChange={(value) => {
+              set_used_weight(value ?? 0);
+              form.setFieldsValue(
+              {
+                used_weight: value ?? 0
+              });
```

Comment:
Create a function in here that calls both set_used_weight and form.setFieldsValue so that you don't have to calculate the value twice.